From 28479934f26bcf9ddeb94125e05ddc5c4312b1f3 Mon Sep 17 00:00:00 2001
Message-Id: <28479934f26bcf9ddeb94125e05ddc5c4312b1f3.1648731282.git.geliang.tang@suse.com>
From: Kuniyuki Iwashima <kuniyu@amazon.co.jp>
Date: Tue, 4 Jan 2022 10:31:49 +0900
Subject: [PATCH] bpf: Add SO_RCVBUF/SO_SNDBUF in _bpf_getsockopt().
Git-commit: 28479934f26bcf9ddeb94125e05ddc5c4312b1f3
Patch-mainline: v5.17-rc1
References: bsn#127
Modified-by-SEL: No


This patch exposes SO_RCVBUF/SO_SNDBUF through bpf_getsockopt().

Signed-off-by: Kuniyuki Iwashima <kuniyu@amazon.co.jp>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Link: https://lore.kernel.org/bpf/20220104013153.97906-3-kuniyu@amazon.co.jp
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 net/core/filter.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/core/filter.c b/net/core/filter.c
index 368fe28c8dc6..cac2be559ab0 100644
--- a/net/core/filter.c
+++ b/net/core/filter.c
@@ -4969,6 +4969,12 @@ static int _bpf_getsockopt(struct sock *sk, int level, int optname,
 			goto err_clear;
 
 		switch (optname) {
+		case SO_RCVBUF:
+			*((int *)optval) = sk->sk_rcvbuf;
+			break;
+		case SO_SNDBUF:
+			*((int *)optval) = sk->sk_sndbuf;
+			break;
 		case SO_MARK:
 			*((int *)optval) = sk->sk_mark;
 			break;
-- 
2.34.1

